pydevices/acq400Devices/acq400_base.py

```diff
@@ -284,18 +320,132 @@ class _ACQ400_TR_BASE(_ACQ400_BASE):
+                #Setting the signal (dX) to use for ACQ2106 stream control
+                uut.s1.TRG       = 'enable'
+                uut.s1.TRG_DX    = 'd1'
+                uut.s1.TRG_SENSE = 'rising'
+
+                #EVENT0 setting in d0:
+                uut.s0.SIG_SRC_TRG_0   = str(self.event0_src.data())
+                uut.s1.EVENT0       = 'enable'
+                uut.s1.EVENT0_DX    = 'd0'
+                uut.s1.EVENT0_SENSE = 'rising'
+                uut.s0.SIG_EVENT_SRC_0 = 'TRG' # In the EVENT bus, the source needs to be TRG to make the transition PRE->POST
+
+            elif str(self.trig_src.data()) in srcs_0:
+                # uut.s0.SIG_SRC_TRG_0   = 'EXT'
+                uut.s0.SIG_SRC_TRG_0   = str(self.trig_src.data())
+                #Setting the signal (dX) to use for ACQ2106 stream control
+                uut.s1.TRG       = 'enable'
+                uut.s1.TRG_DX    = 'd0'
+                uut.s1.TRG_SENSE = 'rising'
+
+                #EVENT0 setting in d1:
+                uut.s0.SIG_SRC_TRG_0   = str(self.event0_src.data())
+                uut.s1.EVENT0       = 'enable'
+                uut.s1.EVENT0_DX    = 'd1'
+                uut.s1.EVENT0_SENSE = 'rising'
+                uut.s0.SIG_EVENT_SRC_0 = 'TRG' # In the EVENT bus, the source needs to be TRG to make the transition PRE->POST
+
+        #If WR is used to trigger, then:
+        else:
+            uut.s1.TRG       = 'enable'
+            uut.s1.TRG_DX    = str(self.trig_dx.data())
+            uut.s1.TRG_SENSE = 'rising'
+
+            uut.s1.EVENT0       = 'enable'
+            uut.s1.EVENT0_DX    = str(self.event0_dx.data())
+            uut.s1.EVENT0_SENSE = 'rising'
+    INIT=init
+
+    def _arm(self):
         import acq400_hapi
         uut = acq400_hapi.Acq400(self.node.data())
         shot_controller = acq400_hapi.ShotController([uut])
+        print("Using HAPI ShotController to run the shot.")
         shot_controller.run_shot()
+        return None
+    _ARM=_arm
+
+    def arm(self):
+        thread = threading.Thread(target = self._arm)
+        thread.start()
     ARM=arm
 
+    def state(self):
+        # State of the Transient recorder NOW (when this method was run)
+        import acq400_hapi
+        uut = acq400_hapi.Acq400(self.node.data())
+        _status = [int(x) for x in uut.s0.state.split(" ")]
+
+        trstate = acq400_hapi.acq400.STATE.str(_status[0])
+        
+        print("TR state: {}".format(trstate))
+        print("TR status: {}".format(_status))
+    STATE=state
+
+    def stop(self):
+        import acq400_hapi
+        uut = acq400_hapi.Acq400(self.node.data())
+        print("PRE {}, POST {} and ELAPSED {}".format(uut.pre_samples(), uut.post_samples(), uut.elapsed_samples()))
+        uut.s0.set_abort=1
+    STOP = stop
 
     def store(self):
+        thread = threading.Thread(target = self._store)
+        thread.start()
+        return None
+    STORE=store
```

Comment:
no need for this.. specify the lowercase name in the METHOD of the action

in general